src/useAsObservableSource.ts

```diff
@@ -0,0 +1,17 @@
+import { observable } from "mobx"
+import { useState } from "react"
+
+import { isPlainObject } from "./utils"
+
+export function useAsObservableSource<T>(current: T): T {
+    if (!isPlainObject(current)) {
+        throw new Error("useAsObservableSource expects an object as first argument")
+    }
+
+    const res = useState(() => observable(current, {}, { deep: false }))[0]
+    if (Object.keys(res).length !== Object.keys(current).length) {
```

Comment:
We should wrap this into a NODE_ENV check just to make it slightly faster in prod.

Reply:
Done